x/escrow/keeper/keeper.go

```diff
@@ -61,25 +68,50 @@ func (k *keeper) AccountCreate(ctx sdk.Context, id types.AccountID, owner sdk.Ac
 		ID:          id,
 		Owner:       owner.String(),
 		State:       types.AccountOpen,
-		Balance:     deposit,
+		Balance:     sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 		Transferred: sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 		SettledAt:   ctx.BlockHeight(),
+		Depositor:   depositor.String(),
+		Funds:       sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 	}
 
 	if err := obj.ValidateBasic(); err != nil {
 		return err
 	}
 
-	if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(deposit)); err != nil {
-		return err
+	if owner.Equals(depositor) {
+		// use owner's account to add balance to escrow account
+		if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(deposit)); err != nil {
+			return err
+		}
+		obj.Balance = deposit
+	} else {
+		if err := k.fetchFundsFromDepositor(ctx, owner, depositor, deposit); err != nil {
+			return err
+		}
+		obj.Funds = deposit
 	}
 
 	store.Set(key, k.cdc.MustMarshal(obj))
 
 	return nil
 }
 
-func (k *keeper) AccountDeposit(ctx sdk.Context, id types.AccountID, amount sdk.Coin) error {
+func (k *keeper) fetchFundsFromDepositor(ctx sdk.Context, owner, depositor sdk.AccAddress, deposit sdk.Coin) error {
+	moduleAcc := k.acctKeeper.GetModuleAccount(ctx, types.ModuleName)
+	if moduleAcc == nil {
+		panic(sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "module account %s does not exist", types.ModuleName))
+	}
+	// use depositor's account to add funds to escrow account
+	_, err := k.authzKeeper.DispatchActions(ctx, owner, []sdk.Msg{&banktypes.MsgSend{
```

Comment:
I think that if we create our own grant object then we may not have to allow sending to the module account.

See for example [here](https://github.com/ovrclk/akash/pull/1270/files#diff-7892d8c61ce49375a51d974cb679b570f806d3d9fe774f14dbcc8d0ab76ff05bR127)

Reply:
I haven't created any separate grant, but just reused the `SendAuthorization` grant provided by cosmos, still allowing the functionality you were looking for, i.e., being able to deposit to the module account without whitelisting it.

Reply:
Okay great - if we don't need to whitelist the module account then this is perfect.

Reply:
Wait - does the authorization allow them to send to *any* address?

Reply:
Yes.

Once a `granter` has granted `SendAuthorization` to a `grantee`, the `grantee` can use the amount specified in the `SendAuthorization` to do whatever they wish to do with it. i.e., They can choose to use it to credit their personal account, or they can use it to pay fees for some transaction, or they can use it to send money to some other random address.

The grant only says that the grantee is allowed to use the allowance, it doesn't restrict "how" to use it.

Reply:
Have added two new sub-commands to deployment module for authorizing deployment payments:
* `akash tx deployment authz grant`: grant authorization for deployment payments
* `akash tx deployment authz revoke`: revoke authorization for deployment payments

Existing authorizations can be queried with the authz module itself like this:
```
# view "deposit deployment" authorizations created by the "funding-wallet" for the "deploy-wallet"
akash query authz grants $(akash keys show funding-wallet -a) $(akash keys show deploy-wallet -a) /akash.deployment.v1beta1.MsgDepositDeployment
```

One can only use `DepositDeploymentAuthorization` to make a deposit for a deployment they own, and for nothing else.